src/main/java/com/xebialabs/overthere/gcp/GcpKeyManagerFactory.java

```diff
@@ -0,0 +1,20 @@
+package com.xebialabs.overthere.gcp;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class GcpKeyManagerFactory {
+
+    private static final ConcurrentMap<String, GcpKeyManager> managers = new ConcurrentHashMap<>();
+
+    private static final GenerateSshKey GENERATE_SSH_KEY = new JCraftGenerateSshKey();
+
+    public static GcpKeyManager create(final String credentialsFile) {
```

Comment:
A potential place for a race condition.

Reply:
@acierto don't understand what is a possible race condition here? In case we go in the if-block we will generate a new gcpKeyManager, and the caller of `create` will use that one. The new gcpKeyManager will not go in the map, but there is no problem, they can both exist in parallel and will be valid: the new one not in the map, will be valid only for this call, and the other one will be reused for next calls of the create with same `credentialsFile`.

Reply:
I am guarding that method with synchronized on `managers`